lisp/mastodon-auth.el

```diff
@@ -60,10 +81,11 @@ if you are happy with unencryped storage use e.g. \"~/authinfo\"."
 
 (defun mastodon-auth--generate-token ()
   "Make POST to generate auth token."
-  (if (or (null mastodon-auth-source-file)
-	  (string= "" mastodon-auth-source-file))
+  (if (and (or (null mastodon-auth-source-file)
+               (string= "" mastodon-auth-source-file))
+           (string= "plain" mastodon-auth-mechanism))
       (mastodon-auth--generate-token-no-storing-credentials)
-    (mastodon-auth--generate-token-and-store)))
+    (mastodon-oauth2--generate-token-and-store)))
```

Comment:
Does that mean we can no longer store the credentials when using `plain` mode?
Does that mean it goes through the token generation for each API call when using `plain`?

Reply:
The OAuth2 library stores the credentials in its own file, which means it doesn't affect the way we store the credentials in `plain` mode. The method will check if the auth mechanism is `plain` or `oauth2`, and will call token generation procedure accordingly.

The OAuth2 library uses the existing tokens, refreshes them if needed, or generates new ones if not available. Hence, the calls will go through the procedure, but HTTP APIs will only be called when necessary.